Orange/data/util.py

```diff
@@ -95,6 +95,15 @@ def hstack(arrays):
         return np.hstack(arrays)
 
 
+def array_equal(a1, a2):
+    """array_equal that supports sparse and dense arrays with missing values"""
+    if sp.issparse(a1):
+        a1 = np.array(a1.todense())
+    if sp.issparse(a2):
+        a2 = np.array(a2.todense())
+    return a1.shape == a2.shape and np.allclose(a1, a2, equal_nan=True)
```

Comment:
I agree.

`csr_matrix`/`csc_matrix` is not necessarily ordered, so it's not trivial to compare. Here's one solution: https://stackoverflow.com/questions/47770906/how-to-test-if-two-sparse-arrays-are-almost-equal

This could probably also be adapted for the case when one matrix is sparse and the other is dense. We can, however, assume that in this case the content is different, too -- at least when making this mistake costs just time, not an error.